assembl/static2/js/app/graphql/IdeaWithPostsQuery.graphql

```diff
@@ -26,6 +28,9 @@ query IdeaWithPosts($id: ID!, $lang: String) {
                 dontUnderstand
                 moreInfo
               }
+              creator {
+                ...AgentProfileInfo
+              }
```

Comment:
Please don't modify this query. subject, creator are retrieved by the Post query.

Reply:
This page display the list of all posts from the idea. They re included in the IdeaWithPosts but are just missing the creator information. Do you suggest that I call another query to fetch all posts with all datas ? It will need to be created as the current PostQuery only fetch a specific post no?

Reply:
Yes, please copy your IdeaWithPosts with a new name and revert the changes you made to it.

Reply:
@vincentfretin I hardly can't do that as the request IdeaWithPosts is used in the idea component and you specified me to reuse that component for brightMirror. If I create a new request, then I must change it in that component but I can't as I don't know if we're in a brightMirror idea before fetching ... Either I keep the idea component and the request as I did, either I create a new "Idea" component with the new query and route brightMirror idea to it. Which one do you prefer?

Reply:
Keep AllIdeasQuery, but use the include directive, see https://graphql.org/learn/queries/#directives

```
--- a/assembl/static2/js/app/graphql/IdeaWithPostsQuery.graphql
+++ b/assembl/static2/js/app/graphql/IdeaWithPostsQuery.graphql
@@ -1,6 +1,7 @@
 #import "./fragments/IdeaMessageColumn.graphql"
+#import "./fragments/AgentProfileInfo.graphql"
 
-query IdeaWithPosts($id: ID!, $lang: String) {
+query IdeaWithPosts($id: ID!, $lang: String, $additionalFields: Boolean!) {
   idea: node(id: $id) {
     ... on Idea {
       id
@@ -15,6 +16,7 @@ query IdeaWithPosts($id: ID!, $lang: String) {
           node {
             ... on Post {
               id
+              subject(lang: $lang) @include(if: $additionalFields)
               parentId
               creationDate
               publicationState
@@ -26,6 +28,9 @@ query IdeaWithPosts($id: ID!, $lang: String) {
                 dontUnderstand
                 moreInfo
               }
+              creator @include(if: $additionalFields) {
+                ...AgentProfileInfo
+              }
             }
           }
```

Reply:
and edit idea.jsx `graphql(IdeaWithPostsQuery, { name: 'ideaWithPostsData' }),` to pass `id` `lang` `additionalFields` variables. additionalFields true for your case, false otherwise.